src/state/Maps/actions.js

```diff
@@ -215,6 +215,42 @@ const removeMapLayer = (mapKey, layerKey) => {
 	};
 };
 
+/**
+ * Remove layers from map
+ * @param mapKey {string}
+ * @param layerKeys {Array}
+ */
+const removeMapLayers = (mapKey, layerKeys) => {
+	return (dispatch, getState) => {
+		if (layerKeys?.length) {
+			const state = getState();
+			let layersToRemove = [];
+
+			layerKeys.forEach(layerKey => {
+				const layerState = Select.maps.getLayerStateByLayerKeyAndMapKey(
```

Comment:
That is just a detail, but selector getLayerStateByLayerKeyAndMapKey is quite expensive. Yes, it should be cached. Sadly we haven't any cheaper selector to check if layer key is in the map.